### PR TITLE
Feature/setup mvc

### PR DESCRIPTION
**Pull request by ngocphong230801**
In this my PR:
- Setup MVC.
- Add localStorage.
- Add constant and helper.

### CODE REVIEW

File: practice/src/scripts/constants/messages.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const handleError = (functionName: string, error: Error): void => {
```
Comment (lintran-agilityio) [inline practice/src/scripts/constants/messages.ts]: what is purpose this function?

File: practice/src/scripts/constants/messages.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    const errorMessage = `Error while ${functionName}: ${error}`;
+    console.error(errorMessage);
```
Comment (lintran-agilityio) [inline practice/src/scripts/constants/messages.ts]: function is not constant folder

File: practice/src/scripts/helpers/display-element.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Toggle the display of an element by changing its style.
+ * @param {string} elementId - The ID of the element to toggle.
+ * @param {boolean} isShow - If true, the element will be displayed, otherwise it will be hidden.
+ */
+export const toggleDisplay = (elementId: string, isShow: boolean): void => {
+    const element = getElementById(elementId);
+
+    // Check if element is not null or undefined before accessing its style property
+    if (element !== null && element !== undefined) {
```
Comment (lintran-agilityio) [inline practice/src/scripts/helpers/display-element.ts]: Is it the same condition `if (element) {}` ?

File: practice/src/scripts/services/localStorage.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import handleError from "../constants/messages";
+
+interface Storage {
+    getTasks: () => any[];
```
Comment (lintran-agilityio) [inline practice/src/scripts/services/localStorage.ts]: why the return type is `any`?

File: practice/src/scripts/services/localStorage.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    saveTasks: (tasks: any[]) => void;
```
Comment (lintran-agilityio) [inline practice/src/scripts/services/localStorage.ts]: the same?

File: practice/src/scripts/models/task.model.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import storage from "../services/localStorage";
+
+interface Task {
```
Comment (lintran-agilityio) [inline practice/src/scripts/models/task.model.ts]: should create type common to re-use

File: practice/src/scripts/constants/messages.ts (hunk 1 above)
Comment (ngocphong230801) [reply practice/src/scripts/constants/messages.ts]: This function is used to notify errors of each feature

File: practice/src/scripts/helpers/messages.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const handleError = (functionName: string, error: Error): void => {
```
Comment (lintran-agilityio) [inline practice/src/scripts/helpers/messages.ts]: should update name "customErrorMessages"

Comment (lintran-agilityio) [inline practice/src/scripts/helpers/messages.ts]: should add comments

File: practice/src/scripts/helpers/messages.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+    const errorMessage = `Error while ${functionName}: ${error}`;
+    console.error(errorMessage);
```
Comment (lintran-agilityio) [inline practice/src/scripts/helpers/messages.ts]: should show alert intead of log

File: practice/src/scripts/services/localStorage.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import { Task } from "./types";
+
+interface Storage {
+    getTasks: () => Task[];
+    saveTasks: (tasks: Task[]) => void;
+}
+
+const storage: Storage = {
+    getTasks: () => {
+        try {
```
Comment (lintran-agilityio) [inline practice/src/scripts/services/localStorage.ts:12]: should cache result and return one time.
Example:
let task = [];
try {
const dataStoraged = localStorage.getItem('tasks');
tasks = dataStoraged ? JSON.parse(dataStoraged) : [];
} catch (error) {
            customErrorMessages('getting tasks', error);
        }

return tasks;

File: practice/src/scripts/services/types.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+export type Task = {
```
Comment (lintran-agilityio) [inline practice/src/scripts/services/types.ts]: it is not services

File: practice/src/scripts/services/localStorage.ts [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+let cachedTasks: Task[] | null = null;
+
+const storage: Storage = {
+    getTasks: () => {
+        try {
+            if (cachedTasks !== null) {
+                return cachedTasks;
+            }
+
+            const tasksJSON = localStorage.getItem('tasks');
```
Comment (lintran-agilityio) [inline practice/src/scripts/services/localStorage.ts]: create storage key constant to reuse

**Review by lintran-agilityio — APPROVED** [review]

**Pull request merged**